test/model/autocomplete_test.dart

```diff
@@ -1,14 +1,166 @@
+import 'dart:async';
+
 import 'package:checks/checks.dart';
+import 'package:fake_async/fake_async.dart';
 import 'package:test/scaffolding.dart';
 import 'package:zulip/api/model/model.dart';
 import 'package:zulip/model/autocomplete.dart';
+import 'package:zulip/model/narrow.dart';
 
 import '../example_data.dart' as eg;
+import 'test_store.dart';
 
 void main() {
+  test('MentionAutocompleteView misc', () async {
+    const narrow = AllMessagesNarrow();
+    final store = eg.store()
+      ..addUsers([eg.selfUser, eg.otherUser, eg.thirdUser]);
+    final view = MentionAutocompleteView.init(store: store, narrow: narrow);
+
+    bool done = false;
+    view.addListener(() { done = true; });
+    view.query = MentionAutocompleteQuery('Third');
+    await Future(() {});
+    check(done).isTrue();
+    check(view.results).single
+      .isA<UserMentionAutocompleteResult>()
+      .has((r) => r.userId, 'userId')
+      .equals(eg.thirdUser.userId);
+  });
+
+  test('MentionAutocompleteView not starve timers', () {
```

Comment:
Co-authored-by: Greg Price <gnprice@zulip.com>

should be greg@, not gnprice@ :)

Reply:
Ah, haha yeah that's right! I've sometimes wondered whether it would have been neater if I took `chris@`, especially if I was the first to break the pattern of using just a first name.